http2-flowcontrol-sim/src/Stream.java

```diff
@@ -0,0 +1,202 @@
+import java.util.ArrayList;
+import java.util.Random;
+
+public class Stream extends SimulationListener implements ClockListener {
+  private int initialWindow;
+  private int windowAvailable;
+  private int publishedWindow;
+  private SimulationClock controller;
+  private int id;
+  private ArrayList<Message> sendQueue;
+  private ClockExecutable sendState;
+  private Connection connection;
+  private int messagesRecieved;
+  private int bufferedBytes;
+
+  public Stream(Connection con, SimulationClock clock) {
+    controller = clock;
+    controller.addListener(this);
+    sendQueue = new ArrayList<Message>();
+    sendState = (int t) -> waitForSend(t);
+    connection = con;
+    messagesRecieved = 0;
+    windowAvailable = connection.availableWindow();
+    initialWindow = windowAvailable;
+    bufferedBytes = 0;
+    id = (new Random()).nextInt();
+
+  }
+
+  @Override
+  public void tick(int t) {
+    sendState.execute(t);
+    System.out.println(this);
+    controller.addEvent(t + connection.latency(), (int time) -> refillWindow());
+    doTickActions(controller);
+  }
+
+  @Override
+  public String toString() {
+    return ("Stream " + id + ": Buffered Bytes " + bufferedBytes + ", Bytes Recieved "
```

Comment:
No need for parens.  You might consider using a StringBuilder, but I think java does this under the hood for the concatenation here.